app/models/task.rb

```diff
@@ -45,17 +49,43 @@ class Task < ApplicationRecord
   scope :min_stars, ->(stars) { average_rating.where('COALESCE(avg_rating, 0) >= ?', stars) }
   scope :sort_by_average_rating_asc, -> { average_rating.order(average_rating: 'ASC') }
   scope :sort_by_average_rating_desc, -> { average_rating.order(average_rating: 'DESC') }
+  scope :fulltext_search, lambda {|input|
+    r = left_outer_joins(:programming_language)
+
+    input.split(/[,\s]+/).each do |keyword|
+      r = r.where(["title ILIKE :keyword
+                    OR description ILIKE :keyword
+                    OR internal_description ILIKE :keyword
+                    OR programming_languages.language ILIKE :keyword
+                    OR EXISTS (SELECT name FROM task_labels JOIN labels ON label_id = labels.id
+                                WHERE task_id = tasks.id
+                                AND name ILIKE :keyword)",
+                   {keyword: "%#{keyword}%"}])
+    end
+    return r
+  }
```

Comment:
As discussed earlier today, my proposal might have shortcomings and change the actual semantics (especially in the case where two labels are assigned to a task and these two labels are used in the fulltext search). We agreed on adding a test case for this scenario first, so that we can test whether the above code will change the behavior.

Reply:
I changed the fulltext_search test case to search for 2 labels instead of just 1. I tested both sql queries and the test case now covers the bug.